### PR TITLE
Remove eBPF probe bundling

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Removes the ability to bundle eBPF probes **for network tracer, OOM Kill check, and TCP Queue Length check** within the system-probe binary.

The functionality is preserved for the runtime-security team.

### Motivation

This is no longer used, as we cannot do this due to licensing restrictions. 

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.

### CODE REVIEW

**Comment by safchain** [comment]
We would prefer to keep it as we use it a lot for local testing, is there any problem with this ?

**Comment by brycekahle** [comment]
> We would prefer to keep it as we use it a lot for local testing, is there any problem with this ?

The bundling already doesn't take into account the `.c` files for runtime compilation. It also is a completely different method of loading from the released agent which has to load the `.o` files from the file system. The agent dev VM sets up a symlink to have the built object files (from `inv system-probe.build`) in the correct location, so `system-probe` reads them by default. Have you tried that?

**Comment by safchain** [comment]
symlinks work locally, but we are using the bundle thing a lot with the functional tests. We build them locally the functional test binary and we copy it on multiple machines.

**Comment by brycekahle** [comment]
Could we make it easy to build a tarball to copy to other machines for testing? Would that solve your use case?

**Comment by lebauce** [comment]
@brycekahle I am afraid of losing a couple hours just to figure out that the .o are not up to date. Right now, I simply have a static testsuite binary with everything inside that I can copy on multiple VMs

**Comment by brycekahle** [comment]
@safchain @lebauce PR updated to keep bundling for runtime-security.

File: tasks/system_probe.py [hunk 1]
```diff
@@ -85,10 +82,9 @@ def build(
                 maj_ver=maj_ver, min_ver=min_ver, patch_ver=patch_ver, target_arch=windres_target
             )
         )
-    else:
-        if compile_ebpf:
-            # Only build ebpf files on unix
-            build_object_files(ctx, bundle_ebpf=bundle_ebpf)
```
Comment (ISauve) [inline tasks/system_probe.py:91]: Could we remove `bundle_ebpf` from here https://github.com/DataDog/datadog-agent/blob/a5fac57539a6e4a0f5695d64a0bf12d0864fd444/tasks/system_probe.py#L51
and remove the bundle tag from the builds tags here? https://github.com/DataDog/datadog-agent/blob/a5fac57539a6e4a0f5695d64a0bf12d0864fd444/tasks/system_probe.py#L95-L96

Comment (brycekahle) [reply tasks/system_probe.py:91]: We have to leave bundling in for the runtime-security team.

File: tasks/system_probe.py [hunk 2]
```diff
@@ -182,7 +145,7 @@ def test(
         clang_tidy(ctx)
 
     if not skip_object_files and not windows:
-        build_object_files(ctx, bundle_ebpf=bundle_ebpf)
```
Comment (ISauve) [inline tasks/system_probe.py:185]: Similarly in this task - can we remove `bundle_ebpf` from the definition & always include the bundle tag?

File: tasks/system_probe.py (hunk 1 above)
Comment (brycekahle) [reply tasks/system_probe.py:91]: > always add the bundle tag to the build tags here?

We don't want to include the bundle tag for system-probe builds unless requested (which is only the runtime-security folks now).

**Review by ISauve — APPROVED** [review]

**Pull request merged**